code/script.js

```diff
@@ -262,82 +270,148 @@ const selectQuestion = () => {
+
   if (group === 'accessories') {
     if (keep) {
       alert(
-        `Yes, the person wears ${attribute}! Keep all that wears ${attribute}`
+        `Yes, the person wears ${attribute}! Keep all persons that wears ${attribute}.`
       )
     } else {
       alert(
-        `No, the person doesn't wear ${attribute}! Remove all that wears ${attribute}`
+        `No, the person doesn't wear ${attribute}! Remove all persons that wears ${attribute}.`
       )
     }
   } else if (group === 'other') {
-    // Similar to the one above
+    if (keep) {
+      alert(
+        `Yes, the person smokes. Keep all smokers.`
+      )
+    } else {
+      alert(
+        `No, the person doesn't smoke. Remove all smokers`
+      )
+    }
   } else {
     if (keep) {
-      // alert popup that says something like: "Yes, the person has yellow hair! Keep all persons with yellow hair"
+      if (currentQuestion.attribute === "hairColor") {
+        alert(
+          `Yes, the person has ${value} hair! Keep all persons with ${value} hair!`
+        )
+      } else {
+        alert(
+          `Yes, the person has ${value} eyes! Keep all persons with ${value} eyes!`
+        )
+      }
     } else {
-      // alert popup that says something like: "NO, the person doesnt have yellow hair! Remove all persons with yellow hair"
+      if (currentQuestion.attribute === "hairColor") {
+        alert(
+          `No, the person doesn't have ${value} hair! Remove all persons with ${value} hair!`
+        )
+      } else {
+        alert(
+          `No, the person doesn't have ${value} eyes! Remove all persons with ${value} eyes!`
+        )
+      }
     }
   }
 
   // filter to keep or remove based on the keep variable.
-  /* charactersInPlay = charactersInPlay.filter((person) => person[attribute] === value)
-    or 
-    charactersInPlay = charactersInPlay.filter((person) => person[attribute] !== value) */
-
-  // Invoke a function to redraw the board with the remaining people.
+  if (keep) {
+    charactersInPlay = charactersInPlay.filter((person) => person[attribute] === value)
+  } else {
+    charactersInPlay = charactersInPlay.filter((person) => person[attribute] !== value)
+  }
+  generateBoard()
 }
 
 // when clicking guess, the player first have to confirm that they want to make a guess.
 const guess = (suspect) => {
-  // store the interaction from the player in a variable.
-  // remember the confirm() ?
+  if (confirm("Are you sure? If your guess is wrong the game will be lost!")) {
+    checkMyGuess(suspect)
+  }
+
   // If the player wants to guess, invoke the checkMyGuess function.
 }
 
 // If you confirm, this function is invoked
 const checkMyGuess = (suspect) => {
-  // 1. Check if the suspect is the same as the secret person's name
-  // 2. Set a Message to show in the win or lose section accordingly
-  // 3. Show the win or lose section
-  // 4. Hide the game board
+
+  let message = `Sorry, you lost! The secret person was ${secret.name}.`
```

Comment:
If I understand it right the variable 'message' has the value of 'Sorry, you lost!' from the start? If it is so, then a suggestion to make it more clear could be to have it empty and then get its value in the if and else depending on whether you win or lose.